File: file_templates/cpp/cpp_basic/class.cpp
{% load kdev_filters %}
/*

 {{ license|lines_prepend:" * " }}
 */

#include "{{ output_file_header }}"

{% include "namespace_use_cpp.txt" %}

{% for method in functions %}
{% with method.arguments as arguments %}

{% if method.type %}{{ method.type }} {% endif %}{{ name }}::{{ method.name }}({% include "arguments_types_names.txt" %})
{

}
{% endwith %}
{% endfor %}

File: file_templates/classes/python_basic/class.py
{% load kdev_filters %}
{% block license_header %}
{% if license %}
#

{{ license|lines_prepend:"# " }}
#
{% endif %}
{% endblock license_header %}


class {{ name }}{% if base_classes %}({% for inh in base_classes %}{{ inh.baseType }}{% if not forloop.last %}, {% endif %}{% endfor %}){% endif %}:
    def __init__(self{% for member in members %}, {{ member.name }}{% endfor %}):
        {% for inh in base_classes %}
        {{ inh.baseType }}.__init__(self)
        {% endfor %}


        {% for member in members %}
        self.{{ member.name }} = {{ member.name }}
        {% empty %}
        pass
        {% endfor %}


    {% for method in methods %}

    {% if method.isStatic %}
    @staticmethod
    def {{ method.name }}({% include "arguments_names.txt" %}):
    {% else %}{% if method.arguments %}
    def {{ method.name }}(self{% for arg in method.arguments %}, {{ arg.name }}{% endfor %}):
    {% else %}
    def {{ method.name }}(self):
    {% endif %}{% endif %}
        pass

    {% endfor %}